app/src/main/java/com/practicum/playlistmaker/player/ui/activity/PlayerActivity.kt

```diff
@@ -16,41 +15,41 @@ import com.practicum.playlistmaker.player.ui.models.PlayerScreenState
 import com.practicum.playlistmaker.player.ui.models.TrackMapper
 import com.practicum.playlistmaker.player.ui.view_model.PlayerViewModel
 import com.practicum.playlistmaker.search.domain.models.Track
+import org.koin.android.ext.android.getKoin
+import org.koin.core.parameter.parametersOf
 
 class PlayerActivity : AppCompatActivity() {
-    private lateinit var binding: ActivityPlayerBinding
-    private lateinit var viewModel: PlayerViewModel
+    private var binding: ActivityPlayerBinding? = null
+    private var viewModel: PlayerViewModel? = null
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         binding = ActivityPlayerBinding.inflate(layoutInflater)
 
-        setContentView(binding.root)
+        setContentView(binding?.root)
 
         val track = getCurrentTrack()
-        viewModel = ViewModelProvider(
-            this,
-            PlayerViewModel.factory(track)
-        )[PlayerViewModel::class.java]
 
-        viewModel.observeState().observe(this) {
+        viewModel = getKoin().get { parametersOf(track) }
```

Comment:
Предложу и альтернативный вариант

```suggestion
class PlayerActivity : AppCompatActivity() {
    private var binding: ActivityPlayerBinding? = null
    private val viewModel: PlayerViewModel by viewModel {
        parametersOf(track)
    }

    private lateinit var track: Track

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityPlayerBinding.inflate(layoutInflater)

        setContentView(binding?.root)

        track = getCurrentTrack()

        // viewModel = getKoin().get { parametersOf(track) }

        viewModel.observeState().observe(this) {
            render(it)
        }

```

Делегат `by` обеспечивает так называемый lazy load, но здесь нужно быть осторожным в одном моменте: track (так как он lateinit) должен быть проинициализирован до первого вызова viewModel, а иначе получим crash
Оба подхода корректны

Reply:
Спасибо за совет! Исправил